client/src/components/hero_location/heroLocationCtrl.js

```diff
@@ -19,8 +20,13 @@
       vm.selection = index;
     };
     vm.confirm = function() {
+      var queueHero = profileFactory.getProfile('username');
       //set location of hero to vm.locations[vm.selection]
-      heroFactory.setOrder(vm.locations[vm.selection]);
+      ajaxFactory.setHeroLocation(queueHero, vm.locations[vm.selection])
+        //will be executed if status code is 200-299,
+        .then(function successCallback(response) {
+          heroFactory.setOrder(vm.locations[vm.selection]);
+      });
       $state.go('hero_task');

```

Comment:
$state.go('hero_task'); line 30 should go inside the successCallback function, after line 28